drivers/kw2xrf/kw2xrf_spi.c

```diff
@@ -0,0 +1,277 @@
+
+    KW2XDRF_SPI->PUSHR = SPI_PUSHR_CTAS(1)
+                         | SPI_PUSHR_PCS(1)
+                         | SPI_PUSHR_TXDATA((uint16_t)addr << 8);
+
+    while (!(KW2XDRF_SPI->SR & SPI_SR_RXCTR_MASK));
+
+    restoreIRQ(cpsr);
+    return (uint8_t)KW2XDRF_SPI->POPR;
+}
+
+void kw2xrf_write_iregs(uint8_t addr, uint8_t *buf, uint8_t length)
+{
+    uint8_t i;
+    unsigned int cpsr = disableIRQ();
+
+    while (KW2XDRF_SPI->SR & SPI_SR_TXCTR_MASK);
+
+    KW2XDRF_SPI->PUSHR = SPI_PUSHR_CTAS(1)
+                         | SPI_PUSHR_PCS(1)
+                         | SPI_PUSHR_CONT_MASK
+                         | SPI_PUSHR_TXDATA(((uint16_t)MKW2XDM_IAR_INDEX << 8) | addr);
+
+    while (!(KW2XDRF_SPI->SR & SPI_SR_RXCTR_MASK));
+
+    KW2XDRF_SPI->POPR;
+
+    for (i = 0; i < (length - 1); i++) {
+        while (!(KW2XDRF_SPI->SR & SPI_SR_TFFF_MASK));
+
+        KW2XDRF_SPI->PUSHR = SPI_PUSHR_CTAS(0)
+                             | SPI_PUSHR_CONT_MASK
+                             | SPI_PUSHR_PCS(1)
+                             | SPI_PUSHR_TXDATA(buf[i]);
+
+        while (!(KW2XDRF_SPI->SR & SPI_SR_RXCTR_MASK));
+
+        KW2XDRF_SPI->POPR;
+    }
+
+    while (!(KW2XDRF_SPI->SR & SPI_SR_TFFF_MASK));
+
+    KW2XDRF_SPI->PUSHR = SPI_PUSHR_CTAS(0)
+                         | SPI_PUSHR_PCS(1)
+                         | SPI_PUSHR_TXDATA(buf[i]);
+
+    while (!(KW2XDRF_SPI->SR & SPI_SR_RXCTR_MASK));
+
+    KW2XDRF_SPI->POPR;
+    restoreIRQ(cpsr);
+}
+
+void kw2xrf_read_iregs(uint8_t addr, uint8_t *buf, uint8_t length)
+{
+    uint16_t iaddr = ((uint16_t)(MKW2XDM_IAR_INDEX | MKW2XDRF_REG_READ) << 8) | addr;
+    uint8_t i;
+    unsigned int cpsr = disableIRQ();
+
+    while (KW2XDRF_SPI->SR & SPI_SR_TXCTR_MASK);
+
+    KW2XDRF_SPI->PUSHR = SPI_PUSHR_CTAS(1)
+                         | SPI_PUSHR_PCS(1)
+                         | SPI_PUSHR_CONT_MASK
+                         | SPI_PUSHR_TXDATA(iaddr);
+
+    while (!(KW2XDRF_SPI->SR & SPI_SR_RXCTR_MASK));
+
+    KW2XDRF_SPI->POPR;
+
+    for (i = 0; i < (length - 1); i++) {
+        while (!(KW2XDRF_SPI->SR & SPI_SR_TFFF_MASK));
+
+        KW2XDRF_SPI->PUSHR = SPI_PUSHR_CTAS(0)
+                             | SPI_PUSHR_CONT_MASK
+                             | SPI_PUSHR_PCS(1)
+                             | SPI_PUSHR_TXDATA(0);
+
+        while (!(KW2XDRF_SPI->SR & SPI_SR_RXCTR_MASK));
+
+        buf[i] = (uint8_t)KW2XDRF_SPI->POPR;
+    }
+
+    while (!(KW2XDRF_SPI->SR & SPI_SR_TFFF_MASK));
+
+    KW2XDRF_SPI->PUSHR = SPI_PUSHR_CTAS(0)
+                         | SPI_PUSHR_PCS(1)
+                         | SPI_PUSHR_TXDATA(0);
+
+    while (!(KW2XDRF_SPI->SR & SPI_SR_RXCTR_MASK));
+
+    buf[i] = (uint8_t)KW2XDRF_SPI->POPR;
+    restoreIRQ(cpsr);
+}
+
+/* read / write packet buffer */
+
+radio_packet_length_t kw2xrf_write_fifo(uint8_t *data, radio_packet_length_t length)
+{
+    radio_packet_length_t i;
+    unsigned int cpsr = disableIRQ();
```

Comment:
uff, disabling interrupts for quite a long time will definitely break RIOTs real-time properties, this should be avoided! Also it doesn't really hurt to jump into a interrupt routine while transferring data on the SPI bus when you are master (and therefore controlling the clock), so why disable it?
(same goes for all your SPI transfer functions!)